src/ofxstatement/statement.py

```diff
@@ -158,9 +159,11 @@ def generate_transaction_id(stmt_line):
     This function can be used in statement parsers when real transaction id is
     not available in source statement.
     """
-    return str(abs(hash((stmt_line.date,
-                         stmt_line.memo,
-                         stmt_line.amount))))
+    h = sha1()
+    h.update(stmt_line.date.strftime("%Y-%m-%d %H:%M:%S").encode("utf8"))
+    h.update(stmt_line.memo.encode("utf32"))
+    h.update(str(stmt_line.amount).encode("utf8"))
```

Comment:
Hm, never heard of this. In any case, if any of unicode characters cannot be really expressed as utf-8, we cannot produce any reasonable output, because ofx file we output is encoded in utf-8. So I'd rather stick with one encoding for that matter.

Reply:
Makes sense, changed